test/test_integration_cluster.rb

```diff
@@ -112,117 +91,166 @@ def test_term_worker_clean_exit
     Process.kill :TERM, pid
     Process.wait pid
 
-    zombies = clean_exit_pids worker_pids
+    zombies = bad_exit_pids worker_pids
 
     assert_empty zombies, "Process ids #{zombies} became zombies"
   end
 
-  # mimicking stuck workers, test respawn with external SIGTERM
+  # mimicking stuck workers, test respawn with external TERM
   def test_stuck_external_term_spawn
-    worker_respawn { |l, phase0_worker_pids|
-      phase0_worker_pids.each { |p| Process.kill :TERM, p }
-    }
+    skip_unless_signal_exist? :TERM
+
+    worker_respawn(0) do |phase0_worker_pids|
+      last = phase0_worker_pids.last
+      phase0_worker_pids.each do |pid|
+        Process.kill :TERM, pid
+        sleep 4 unless pid == last
```

Comment:
Why 4? Please leave a comment.

Reply:
I say this because ocasionally in integration tests we have to sleep specific times for specific reasons, so if there are reasons, they should be documented.